src/cpp/src/speculative_decoding/speculative_decoding_impl.hpp

```diff
@@ -0,0 +1,87 @@
+// Copyright (C) 2023-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "openvino/genai/continuous_batching_pipeline.hpp"
+#include "continuous_batching_impl.hpp"
+#include "continuous_batching_for_speculative_decoding_impl.hpp"
+
+namespace ov::genai {
+
+class SpeculativeDecodingMetrics {
+    using AcceptanceRate = std::vector<float>;
+    // { request_id, acceptance_rate }
+    std::map<int64_t, AcceptanceRate> m_acceptance_rate;
+
+public:
+    float get_avg_acceptance_rate(int64_t request_id = 1) {
```

Comment:
let's move implementation of the methods to .cpp files

Reply:
Fixed